exporters/prometheus/exporter.go

```diff
@@ -233,3 +233,51 @@ func sanitizeRune(r rune) rune {
 	}
 	return '_'
 }
+
+func sanitizeName(n string) string {
+	// This algorithm is based on strings.Map from Go 1.19.
+	const (
+		replacement = '_'
+		width       = 1
+	)
+
+	valid := func(i int, r rune) bool {
+		// Taken from
+		// https://github.com/prometheus/common/blob/dfbc25bd00225c70aca0d94c3c4bb7744f28ace0/model/metric.go#L92-L102
+		if (r >= 'a' && r <= 'z') || (r >= 'A' && r <= 'Z') || r == '_' || r == ':' || (r >= '0' && r <= '9' && i > 0) {
+			return true
+		}
+		return false
+	}
+
+	// This output buffer b is initialized on demand, the first time a character
+	// needs to be replaced.
+	var b strings.Builder
+	for i, c := range n {
+		if valid(i, c) {
+			continue
+		}
+		b.Grow(len(n))
+		b.WriteString(n[:i])
+		b.WriteRune(replacement)
+		n = n[i+width:]
+		break
+	}
```

Comment:
Handle prefix and multi byte characters:

```suggestion
	// This algorithm is based on strings.Map from Go 1.19.
	const replacement = '_'

	valid := func(i int, r rune) bool {
		// Taken from
		// https://github.com/prometheus/common/blob/dfbc25bd00225c70aca0d94c3c4bb7744f28ace0/model/metric.go#L92-L102
		if (r >= 'a' && r <= 'z') || (r >= 'A' && r <= 'Z') || r == '_' || r == ':' || (r >= '0' && r <= '9' && i > 0) {
			return true
		}
		return false
	}

	// This output buffer b is initialized on demand, the first time a
	// character needs to be replaced.
	var b strings.Builder
	for i, c := range n {
		if valid(i, c) {
			continue
		}

		if i == 0 && c >= '0' && c <= '9' {
			// Prefix leading number with replacement character.
			b.Grow(len(n) + 1)
			b.WriteByte(byte(replacement))
			break
		}
		b.Grow(len(n))
		b.WriteString(n[:i])
		b.WriteByte(byte(replacement))
		width := utf8.RuneLen(c)
		n = n[i+width:]
		break
	}
```